### ISSUE TITLE
Reviewing Global Styles main panel

### ISSUE THREAD

**Issue by pablohoneyhoney** [WordPress/gutenberg#38934]
With GS live, it's pertinent to take a look at how elements are presented in the interface compared to the original intent. There are likely valid reasons (technical, functional, etc.) for certain objects to evolve and be at the state they are, while some do need further advancement to depict the best experience and UI possible. 

Generally, it's in a fantastic state and it's already opening up new avenues of thinking. It is worth however to look at the foundational parts to be able to grow healthier. The spacings, alignments, and scales are the main aspects that could use some help. 

The intent of this issue is to itemize and discuss certain improvements in the root panel, and it will be useful to later review other areas of the full styles experience. It is not the intent to look at larger functional changes like styles switching or how colors get represented in the preview from the theme –that's why I chose to use a single color to cover the below. 

### Current vs. Original Design

<img width="689" alt="Screen Shot 2022-02-20 at 1 23 26 PM" src="https://user-images.githubusercontent.com/25516611/154842279-98581cba-6619-4637-b29b-854a66725d36.png">

- [x] **Adjust styles preview box size.** #39978
Current box is 248x150 while the original was 248x152, which matches more precisely a grid of 8 and provides enough white space for typographies of different structure.  
 
<img width="731" alt="Screen Shot 2022-02-20 at 12 22 24 PM" src="https://user-images.githubusercontent.com/25516611/154840156-90faacb2-385e-4e1f-b405-f255949e27a5.png">

- [x] **Adjust color swatches size.** #39978
Current swatch is 40x40 while the original was 32x32, which creates a more balanced presence with the typography as they deserve equal representation as a style preview. The current example (left) is quite bulky. 

<img width="557" alt="Screen Shot 2022-02-20 at 12 33 31 PM" src="https://user-images.githubusercontent.com/25516611/154840450-6ff13b3f-683d-4d20-a2ad-66b19de39b4c.png">

- [x] **Smooth the corners of the styles box.** #39980 and #40032
It might be well a browser issue, but it persists for me both in Safari and Brave. See below the corrupted corner. 

<img width="513" alt="Screen Shot 2022-02-20 at 12 46 06 PM" src="https://user-images.githubusercontent.com/25516611/154840922-8e0fb69b-cd8c-4f36-9c28-cebdad8c9c7c.png">

- [x] **Correct labels colors.** #40055, #40113

Worth reviewing them all but the elements seem to have a transparency or a slight grey. If I'm not mistaken the color code used is: `rgba(0, 0, 0, 0.847)`. The original idea was to leverage the G2 dark grey `rgba(30, 30, 30)` or `#1E1E1E` as the principle dark color, which might be a different topic all together. But at least labels are better off without that grey or transparent look so they align with the rest of elements in the interface, like the icons. 

<img width="274" alt="Screen Shot 2022-02-20 at 12 58 46 PM" src="https://user-images.githubusercontent.com/25516611/154841341-97eb3e48-c65b-445b-96db-93109df1d009.png">

- [x] **Align icons with labels.** #39982

There's a slight horizontal unbalance between the icon and label that is worth looking at, as it might be an issue with the icon set up or the label position. That is also applicable to the relationship between block and arrow. 

<img width="270" alt="Screen Shot 2022-02-20 at 1 00 37 PM" src="https://user-images.githubusercontent.com/25516611/154841427-43ec6804-ba61-4add-8906-490f266fd392.png">

- [ ] **Consider removing the heading separator.**

This might be a topic to look at as we look at the whole flow, since this separator is likely inherited from the main component and it's also perhaps needed in interior sections. Anyhow, the separator isn't needed in the main panel when it lives with the style preview. This can also bring up the preview a bit higher in the panel. 

<img width="1067" alt="Screen Shot 2022-02-20 at 1 07 28 PM" src="https://user-images.githubusercontent.com/25516611/154841653-7581c9c6-0565-4ad8-ba04-0db6c8e7e277.png">

It also causes however some frictions currently when the element represents its own separator (as below). 

<img width="295" alt="Screen Shot 2022-02-20 at 1 02 54 PM" src="https://user-images.githubusercontent.com/25516611/154841669-7b3bc86f-b7b2-44ec-9ac5-1334e00155b3.png">

- [x] **Review spacings between all _elements_** #40526 #40588 #40592

While there's no need to match the original spacing, the current has all elements fairly expanded and could use some tighting up for better visual grouping and generally reduce the height of the panel elements. If it helps for reference, the original intent had a padding of 16, but the one live. While correct, it also includes the separator component own padding, which increases significantly the spacing. 

<img width="1025" alt="Screen Shot 2022-02-20 at 1 21 48 PM" src="https://user-images.githubusercontent.com/25516611/154842201-d1685fc6-1533-4539-bae9-b323d6bb7caa.png">

The spacing discordance also applies to styles elements. 

<img width="975" alt="Screen Shot 2022-02-20 at 1 30 54 PM" src="https://user-images.githubusercontent.com/25516611/154842575-ad805015-ac11-40b3-80f3-f21e7cf845a7.png">

**Comment by ciampo** [comment]
Hey @pablohoneyhoney, here are my considerations (and a few questions) after a brief investigation:

> ### Adjust styles preview box size

This can be improved as you suggest with quick adjustment directly to the Global Styles [preview](https://github.com/WordPress/gutenberg/blob/6b6b7980aec9e457c86b5ef7df0f148e8caa07d7/packages/edit-site/src/components/global-styles/preview.js) in the sidebar.

> ### Adjust color swatches size

This is also a quick adjustment that can be made similarly to the previous one.

When I look at what we currently have on `trunk`, [the size of the color swatch appears to be `30`](https://github.com/WordPress/gutenberg/blob/6b6b7980aec9e457c86b5ef7df0f148e8caa07d7/packages/edit-site/src/components/global-styles/preview.js#L136-L139) (and not `40`). There may have been recent changes, compared to when the screenshots above were taken — this is what I see as of today when loading the latest version of the Global Styles sidebar on my local machine:

![Screenshot 2022-04-01 at 12 40 00](https://user-images.githubusercontent.com/1083581/161248028-5de51509-48d8-4392-8d29-1d975f79719a.png)

I'm happy to update the size of the color swatch to `32`, but **I wonder if the rest of the spacing (and the font size) would also need updating?** It would be useful to have a complete design spec for the Styles Preview card

> ### Smooth the corners of the styles box

Wrapping [the `<StylesPreview />` component](https://github.com/WordPress/gutenberg/blob/7697a1cdd4dd94b2363635bba06f37d4d8674cad/packages/edit-site/src/components/global-styles/screen-root.js#L40) inside `<CardMedia />` should definitely improve the situation (🤞 ), although there's always the chance that browsers rendering engines can introduce some sub-pixel / anti-aliasing artefacts.

> ### Correct labels colors

> Worth reviewing them all but the elements seem to have a transparency or a slight grey. If I'm not mistaken the color code used is: `rgba(0, 0, 0, 0.847)`.

**I wasn't able to find elements with such transparent color, these elements seem to be black (`#000000`) for me.**

> The original idea was to leverage the G2 dark grey `rgba(30, 30, 30)` or `#1E1E1E` as the principle dark color

I can see the `#1e1e1e` color [being applied to all contents of "complementary area"](https://github.com/WordPress/gutenberg/blob/c1ea8d60db9f05ebbfbaced50a88bfe30a7a0270/packages/interface/src/components/complementary-area/style.scss#L3) (a part of the codebase that I never worked on before). But currently, there are a few style rules with higher specificity that are basically applying the black color to all content — one of them is [this style rule from the `Surface` component](https://github.com/WordPress/gutenberg/blob/e108be7e84dd2045d87cf2bd3ca52f58a83217c7/packages/components/src/surface/styles.js#L13).

I'd be reluctant to make changes directly in the source components — instead, we can look into applying style overrides specifically to the global styles sidebar in order to change the text color.

> ### Align icons with labels

This is an adjustment that can be easily applied as a "global" adjustment to all icons — it would be more complicated if we would need to tweak the alignment on an icon-by-icon basis.

> ### Consider removing the heading separator

As you also mentioned, the header used in the Global Styles sidebar is a shared component called [`ComplementaryAreaHeader`](https://github.com/WordPress/gutenberg/blob/6c4eb8d173028ce5f7c149d4a08045252cdd44c9/packages/interface/src/components/complementary-area-header/index.js) that lives in a separate package (`@wordpress/interface`).

The `ComplementaryAreaHeader` component gets most of its styles (including [the bottom border](https://github.com/WordPress/gutenberg/blob/fcc32eeee7901472279b609a7d35b18c963388e8/packages/components/src/panel/style.scss#L39)) by using the `components-panel__header` classname (instead of using the actual `PanelHeader` component from `@wordpress/components` — this by itself is a dangerous anti-pattern!).

In short, we can't remove the border styles from the "source" (i.e. the styles of the [`PanelHeader`](https://github.com/WordPress/gutenberg/blob/9c09a52f34822c61b264d25f3e8e87c3f3a2d3c5/packages/components/src/panel/README.md#panelheader) component). 

I see two choices — both feel slightly hacky, and I don't personally know the `@wordpress/interface` package well enough to be certain about what's the best one:

- manually override the styles of the header (and remove the border) only in the Global Styles sidebar
- add a prop to `ComplementaryArea` to be able to disable the header's bottom border

Maybe @jorgefilipecosta  can help here?

> ### Review spacings between all _elements_

The current spacing is a result of different paddings (mostly `Card` and `ItemGroup`) and a margin of `1.5em` on the separator (which again, comes from [the styles applied by the "complementary area"](https://github.com/WordPress/gutenberg/blob/c1ea8d60db9f05ebbfbaced50a88bfe30a7a0270/packages/interface/src/components/complementary-area/style.scss)).

Adjustments can be definitely made. It would be interesting to see which of these adjustments can happen at the "source" and which need to be applied as a style override specifically for Global Styles


---

I'll follow up shortly with a few PRs tackling the points highlighted above.

**Comment by annezazu** [comment]
[During the 6.0 walkthrough](https://make.wordpress.org/core/2022/03/30/6-0-product-walk-through/), the prominence of switching styles was raised and I wanted to flag here since this seems to capture an overview of current refinements to the Global Styles panel:

<img width="283" alt="Screen Shot 2022-04-05 at 6 40 32 PM" src="https://user-images.githubusercontent.com/26996883/161878529-61c96788-df72-44ad-8b7c-52a7ab9a003f.png">

Currently, the "Browse styles" matches the design of the "Blocks" section below it, when perhaps it should be fairly distinct. I'll leave it to you all to figure out how best to proceed but wanted to close the loop here as it seems that the above designs existed before style variations were being worked on. cc @critterverse who has been working on the animation and previews.  If you'd prefer this be an additional issue, I'm happy to open one. 

Of note, you can use [this gist](https://gist.github.com/kjellr/1307387f8d97e84662c4b07af873a896) (and the instructions in the comment) to recreate having style options locally.

**Comment by ciampo** [comment]
I'll let @pablohoneyhoney and @jasmussen digest the feedback here, including whether this should be handed in a separate issue

**Comment by pablohoneyhoney** [comment]
Apologies @ciampo mind clarifying what you mean with this – I'm not sure I understand or link it to your notes above:

> including whether this should be handed in a separate issue

Regarding the rest, all of those suggestions and tweaks are already good progress and worth doing as a next iteration. We can keep this issue open for further improvement notes, or create a separate one once all is addressed or nuanced. 

> Currently, the "Browse styles" matches the design of the "Blocks" section below it, when perhaps it should be fairly distinct. 

I agree with this @annezazu and believe I mentioned it in the context of @critterverse iterations in that area.

**Comment by ciampo** [comment]
> mind clarifying what you mean with this – I'm not sure I understand or link it to your notes above:
>
> > including whether this should be handed in a separate issue

Sure! My reasoning here is that @annezazu's [feedback](https://github.com/WordPress/gutenberg/issues/38934#issuecomment-1089642706) is still quite at a high-level (differentiate the design of the "Browse styles" and the "Blocks" sections), and so I proposed that we may move it to a separate GitHub issue, where folks can discuss and work on a design spec until it's ready to be executed and tracked separately from the initial list of tweaks already suggested in this issue.

Relatively to this issue, @mirka and I have been progressively addressing most of the feedback. The only areas that we haven't tackled yet are:

- _"Consider removing the heading separator"_ (this is a non-trivial fix — I left some more details in [this comment](https://github.com/WordPress/gutenberg/issues/38934#issuecomment-1085804088), including a ping to @jorgefilipecosta in case he had any suggestions)
- _"Review spacings between all elements"_ (not trivial, since we'll need to balance changes to the core components and tweaks specific to what's called the ["complementary area"](https://github.com/WordPress/gutenberg/blob/c1ea8d60db9f05ebbfbaced50a88bfe30a7a0270/packages/interface/src/components/complementary-area/style.scss) and the sidebar)

**Comment by jasmussen** [comment]
Agreed there are lots of small fixes we can make that will benefit the UI as it exists today and for the foreseeable future.

One additional consideration is how global styles can evolve in the future. For example, #36667#issuecomment-1087711883 suggests moving it to the left on a dark background and shown in context of other site-wide tools such as "Navigation". That would absorb the "Templates" item as well. Would that make room for "Browse styles" to live as a normal menu item next to Color and Typography, perhaps even Blocks? Or do they still need sectioning off and/or live in the context of the style card?

**Comment by critterverse** [comment]
Shared a few ideas for iterating on the Browse styles button in https://github.com/WordPress/gutenberg/issues/40478 :)

**Comment by jasmussen** [comment]
One quick note worth considering. The borderless heading as designed looks great and makes the whole panel feel a bit more like a cohesive unit.

Actually removing that border might require a little extra effort, however. On very small viewports, or as soon as you click into a sub-section, the header stays fixed at the top and a scrollbar can appear — in those cases keeping the border might be useful:
<img width="523" alt="Screenshot 2022-04-22 at 08 18 32" src="https://user-images.githubusercontent.com/1204802/164615523-fa572f81-104c-46b7-a0ba-0444b8ed6124.png">

The borderless look works okay with scrollbars in the following experiment, but it might work better if the border would appear contextually with a scrollbar:

![experiment](https://user-images.githubusercontent.com/1204802/164615868-65f0d714-2285-4d49-8f9d-2f4b57855e14.gif)

Given that #36667 explores global styles on a dark background, it _may_ not be worth exploring such a border-behavior right at this point.

**Comment by ciampo** [comment]
> it may not be worth exploring such a border-behavior right at this point.

I agree — removing the border from the heading feels still very much an idea that needs more exploration before it can be implemented (plus, it wouldn't be a trivial change, as explained [before](https://github.com/WordPress/gutenberg/issues/38934#issuecomment-1085804088)).

I am in favour of delaying the work on this potential change until it feels ready for implementation.

**Comment by jorgefilipecosta** [comment]
Hi @ciampo,

> "Consider removing the heading separator" (this is a non-trivial fix — I left some more details in https://github.com/WordPress/gutenberg/issues/38934#issuecomment-1085804088, including a ping to @jorgefilipecosta in case he had any suggestions)

For this task, I think the solution "manually override the styles of the header (and remove the border) only in the Global Styles sidebar" is the right one. We have a class for the global styles sidebar we can use specific styles to customize that sidebar specifically because it has very specific needs.

**Comment by ciampo** [comment]
Update:

From the [original issue description](https://github.com/WordPress/gutenberg/issues/38934#issue-1145007194), @mirka and I **tackled all points except the one about removing the heading separator**. I kept the original description up to date with references to all of the PRs that we merged for each point.

Regarding the heading separator, I agree with @jasmussen 's [message](https://github.com/WordPress/gutenberg/issues/38934#issuecomment-1106055004) — in short:

> it may not be worth exploring such a border-behavior right at this point.

This behaviour should be implemented (ideally by someone more knowledgeable about this part of the codebase than myself) only once it's in a more polished state design-wise.

> For this task, I think the solution "manually override the styles of the header (and remove the border) only in the Global Styles sidebar" is the right one. We have a class for the global styles sidebar we can use specific styles to customize that sidebar specifically because it has very specific needs.

I agree that, in case we wanted to implement this change, manually overriding the header styles only in the Global Styles sidebar would be the quickest way. At the same time, it feels like we should still try to tidy these styles up more at the root (i.e. `ComplementaryArea`), rather than adding one more layer of style overrides.

In the light of the above, **I believe that we can close this issue:**

- "Browse styles" button explorations are already tracked in #40478
- #36667 explores global styles on a dark background
- We can open a new issue specifically around the removal of the Global Styles heading separator if/when we see fit

---

_Side note: in the upcoming weeks I'm going to try to partially rewrite `ComplementaryAreaHeader` so that it doesn't use [hardcoded classnames from the `Panel` component](https://github.com/WordPress/gutenberg/blob/6c4eb8d173028ce5f7c149d4a08045252cdd44c9/packages/interface/src/components/complementary-area-header/index.js#L37) — these changes won't really be related to what is being discussed in this PR, it's mostly a refactor to avoid this anti-pattern._
